infer/modules/train/extract/extract_f0_print.py

```diff
@@ -1,175 +1,170 @@
-                )
-        elif f0_method == "harvest":
-            f0, t = pyworld.harvest(
-                x.astype(np.double),
-                fs=self.fs,
-                f0_ceil=self.f0_max,
-                f0_floor=self.f0_min,
-                frame_period=1000 * self.hop / self.fs,
-            )
-            f0 = pyworld.stonemask(x.astype(np.double), f0, t, self.fs)
-        elif f0_method == "dio":
-            f0, t = pyworld.dio(
-                x.astype(np.double),
-                fs=self.fs,
-                f0_ceil=self.f0_max,
-                f0_floor=self.f0_min,
-                frame_period=1000 * self.hop / self.fs,
-            )
-            f0 = pyworld.stonemask(x.astype(np.double), f0, t, self.fs)
-        elif f0_method == "rmvpe":
-            if hasattr(self, "model_rmvpe") == False:
-                from rvc.f0.rmvpe import RMVPE
-
-                print("Loading rmvpe model")
-                self.model_rmvpe = RMVPE(
-                    "assets/rmvpe/rmvpe.pt", is_half=False, device="cpu"
-                )
-            f0 = self.model_rmvpe.compute_f0(x, filter_radius=0.03)
-        return f0
-
-    def coarse_f0(self, f0):
-        f0_mel = 1127 * np.log(1 + f0 / 700)
-        f0_mel[f0_mel > 0] = (f0_mel[f0_mel > 0] - self.f0_mel_min) * (
-            self.f0_bin - 2
-        ) / (self.f0_mel_max - self.f0_mel_min) + 1
-
-        # use 0 or 1
-        f0_mel[f0_mel <= 1] = 1
-        f0_mel[f0_mel > self.f0_bin - 1] = self.f0_bin - 1
-        f0_coarse = np.rint(f0_mel).astype(int)
-        assert f0_coarse.max() <= 255 and f0_coarse.min() >= 1, (
-            f0_coarse.max(),
-            f0_coarse.min(),
+from torch.multiprocessing import Process, set_start_method
+
+set_start_method("spawn", force=True)
+
+
+# A helper function to log in both the terminal and the logfile
+def __log(logfile, data: str) -> None:
+    logger.info(data)
+    logfile.write(f"{data}\n")
+
+
+def save_f0(
+    predictor: F0Predictor,
+    inp_path: str,
+    coarse_path: str,
+    feature_path: str,
+    logfile: str,
+) -> None:
+    """
+    Compute the F0 and save the results in a log file
+
+    Args:
+        inp_path: str - Input path for the audio
+        coarse_path: str - Path to save f0 coarse to
+        feature_path: str - Path to save f0 features to
+        logfile: str - The main log file of the f0 extraction
+    """
+
+    try:
+        # So fun fact:
+        # As it turns out, you just can't pickle the TextIOWrappers, for some reason?
+        # Why is that? Who knows! But this unfortunately means that we have to
+        # Open the logfiles within each thread in order to have anything done here...
+        # This makes the code I/O bound and there's not a lot to do about it, i think
+        __logfile = open(logfile, "w")
+
+        out_features = predictor.compute_f0(load_audio(inp_path, 16000))
+        out_coarse = 0  # TODO: add coarse f0

```

Comment:
Have you done this?

Reply:
> Have you done this?

Technically it works, yeah
However I'm having troubles saving features now, for some reason...

Working on it!

Reply:
Now that features are saving properly, the same coarse function is being called on the `out_features`
707536e